v2/pkg/operators/common/dsl/dsl.go

```diff
@@ -809,6 +811,66 @@ func init() {
 			data := gcm.Seal(nonce, nonce, []byte(value), nil)
 			return data, nil
 		}),
+		"generate_jwt": makeDslFunction(3, func(args ...interface{}) (interface{}, error) {
+			var claims jwt.MapClaims
+
+			jsonData := []byte(args[0].(string))
+			signature := []byte(args[1].(string))
+			algorithm := args[2].(string)
+
+			// Valid algorithms are: RS256, HS256, RS384, RS512, PS384, ES256, EdDSA, HS384, HS512, PS256, ES384, ES512, PS512.
+			// Use github.com/golang-jwt/jwt/v4.GetAlgorithms().
+
+			signingMethod := jwt.GetSigningMethod(algorithm)
+			if signingMethod == nil || signingMethod == jwt.SigningMethodNone {
+				return nil, fmt.Errorf("invalid algorithm: %s", algorithm)
+			}
+
+			err := json.Unmarshal(jsonData, &claims)
+			if err != nil {
+				return nil, err
+			}
+
+			token := jwt.NewWithClaims(signingMethod, claims)
+
+			tokenString, err := token.SignedString(signature)
+			if err != nil {
+				return nil, err
+			}
+
+			return tokenString, nil
```

Comment:
@dwisiswant0 , I think we are missing a attack vector here (i.e None algorithm attack).
This is a popular attack vector where some jwt vendors do not verify if algorithm is `noNe` which allows to create valid jwt.

I propose that  if algorithm is `none,None` or any mixedcase variations we can create a jwt without signature using given details (i.e base64 encode header and body). 

### Implementation
we can check if given algorithm is supported if not convert it to lowercase and trim and check if it none .
if yes using original algorithm value create jwt without signature

Reply:
@tarunKoyalwar - Initially, I created the function with an _optionalAlgorithm_ input, which if not provided will use `alg="none"` to generate the token. However, it seems that it is not explained in the lib docs how to use it (track: https://github.com/golang-jwt/jwt/issues/264). While waiting for an answer or looking for an alternative way for this case, I will mark this PR as a draft - therefore I'll revert the input algorithm not to be required but optional.